Makefile.cflags

```diff
@@ -66,3 +66,20 @@ endif
 ifeq ($(origin ARFLAGS),default)
   ARFLAGS = rcs
 endif
+
+# check if we want to make a build without any optimization
+ifeq (1,$(ALL_DEBUG_BUILD))
+  CFLAGS += -g3 -gdwarf-3
+
+  # we may only disable optimization if it's not prohibited
+  ifneq (1,$(MUST_OPTIMIZE))
+    # check if the compiler supports optimize for debug
+    ifeq ($(shell $(CC) -Og -E - 2>/dev/null >/dev/null </dev/null; echo $$?),0)
```

Comment:
> independent of the idea behind, you could use &>/dev/null instead of 2>/dev/null >/dev/null

I'm not so sure about that... dash (which is used by Ubuntu for `/bin/sh` e.g.) does not support the `&>` operator.